lib.php

```diff
@@ -69,21 +70,29 @@ function local_smartmedia_pluginfile($course, $cm, $context, $filearea, $args, $
     $utility = new \local_smartmedia\utility();
     $args = $utility->update_args($args);
 
-    $itemid = $args[0];
+    $itemid = array_shift($args);
 
     // Make sure the user is logged in and has access to the module.
-    require_login($course);
+    require_login();
```

Comment:
was it intentional to drop the course check here?

Reply:
Yes, course is null, as the files are always stored at system context